BEIMA.Backend/MongoService/Device.cs

```diff
@@ -124,6 +160,9 @@ public BsonDocument GetBsonDocument()
             CheckNullArgument(Notes, nameof(Notes));
             CheckNullArgument(Location, nameof(Location));
```

Comment:
You check whether or not Location/LastModified is null, but Location and LastModified are instantiated when using the empty constructor. 

Please instantiate Location and LastModified back to the full constructor.

Please look at how this is done in Building.cs and User.cs. You should verify that all the fields inside the actual Location/LastModified objects are not null.